src/main_logic/main.cpp

```diff
@@ -59,8 +58,8 @@ int main()
             f(simulation_map, &iteration_number);
         }
 
-        if(!send_particles_to_visualization(visualization_endpoint, simulation_map->nodes,
-                                            simulation_map->get_n_of_nodes()))
+        if(!send_particles_to_visualization(visualization_endpoints, simulation_map->nodes,
+                                            simulation_map->get_n_of_nodes(), polyhedron, polyhedron->get_n_of_faces()))
```

Comment:
You definitely should not give `polyhedron->n_of_faces` as an argument to `send_particles_to_visualization`. It must be retrieved inside of the function body